### PR TITLE
Add TrainTripForm and busTripForm, stored in local storage and visibl…

### PR DESCRIPTION
**Pull request by johdyn**
…e in triplist, update of summary chart and some change in the layout

### CODE REVIEW

File: src/components/AddBusForm.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            required
+          />
+          <input
+            type="date"
+            className="add-trip-datepicker"
+            value={date}
+            pattern="\d{4}-\d{2}-\d{2}"
+            onChange={handleDateChange}
+            required
+          ></input>
```
Comment (hugh-burgess) [inline src/components/AddBusForm.js:77]: just make this self-closing no?

File: src/components/AddBusForm.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          <div className="radio-button-container">
+            <label>
+              <input
+                type="radio"
+                checked={1 === radioButton}
+                onChange={(e) => {
+                  setRadioButton(1);
+                }}
+                name="bus"
+              />{" "}
```
Comment (hugh-burgess) [inline src/components/AddBusForm.js:97]: you can delete that space thingy

File: src/components/AddBusForm.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            </label>
+            <label>
+              <input
+                type="radio"
+                checked={2 === radioButton}
+                onChange={(e) => {
+                  setRadioButton(2);
+                }}
+                name="bus"
+              />{" "}
```
Comment (hugh-burgess) [inline src/components/AddBusForm.js:108]: same here

File: src/components/AddBusForm.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          <p className="add-car-emission-paragraph">
+            Emission in kg:{carbonEstimate.toFixed(2)}
+          </p>
+        ) : null}
+        {showAddButton ? (
+          <Button
+            type="secondary"
+            text="Add to My Trips"
+            onClick={handleAddTrip}
+          ></Button>
```
Comment (hugh-burgess) [inline src/components/AddBusForm.js:128]: if no value then self-closing?

File: src/components/AddTrainForm.js [hunk 5]
```diff
@@ -90,97 +46,83 @@ export default function AddCarTripForm() {
...
-    setSelectVehicleModelID(selectedModel.data.id);
+  function handlePassengerChange(event) {
+    const { value } = event.target;
+    if (value > 0) {
+      setPassengers(value);
+    } else setPassengers(0);
   }
 
   function handleBackClick() {
     history.push("/add-trip");
```
Comment (hugh-burgess) [inline src/components/AddTrainForm.js:57]: if youre last page before this is the add-trip page, then just use `history.goBack()`

File: src/components/CarTripItem.css [hunk 6]
```diff
@@ -4,7 +4,8 @@
   position: relative;
   z-index: 1;
   padding: 1em;
-  height: 80px;
+  min-height: 80px;
+  max-height: auto;
```
Comment (hugh-burgess) [inline src/components/CarTripItem.css:8]: i THINK, this is default but check

File: src/components/DoughnutChart.js [hunk 7]
```diff
@@ -2,24 +2,25 @@ import "./DoughnutChart.css";
...
-              data: [flightSum, carSum],
-              backgroundColor: ["#aaacd1", "#2e3557"],
-              borderColor: ["white", "white"],
+              data: [flightSum, carSum, trainSum, busSum],
+              backgroundColor: ["#aaacd1", "#2e3557", "#3ab09c", "#faffe4"],
+              borderColor: ["transparent"],
             },
           ],
         }}
+        options={{ plugins: { legend: { labels: { color: "whitesmoke" } } } }}
```
Comment (hugh-burgess) [inline src/components/DoughnutChart.js:23]: lol

File: src/components/IconButton.css [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  /* box-shadow: 1px 1px 4px whitesmoke; */
+  /* border-radius: 50px;
+  background: linear-gradient(145deg, #0a7971, #08665f);
+  box-shadow: 20px 20px 60px #08605a, -20px -20px 60px #0a827a; */
+  /* border-radius: 50px;
+  background: linear-gradient(145deg, #0a7971, #08665f);
+  box-shadow: 12px 12px 24px #08605a, -12px -12px 24px #0a827a; */
+  /* border-radius: 50px;
+  background: linear-gradient(145deg, #0a7971, #08665f);
+  box-shadow: 6px 6px 11px #08605a, -6px -6px 11px #0a827a; */
```
Comment (hugh-burgess) [inline src/components/IconButton.css:21]: if unused then delete no?

File: src/components/SummaryCard.js [hunk 9]
```diff
@@ -1,23 +1,30 @@
...
+
   return (
     <div className="summary-card">
+      <input type="date" />
       <select className="period-select-box">
         <option>Choose a month or year</option>
         {/* <option>This month</option>
         <option>This year</option> */}
       </select>
       <h3>Your Total CO2: {totalSum} kg</h3>
```
Comment (hugh-burgess) [inline src/components/SummaryCard.js:20]: Should this maybe be...

```suggestion
      <h3>`Your Total CO2: ${totalSum} kg`</h3>
```

File: src/components/TrainTripItem.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  return (
+    <article className="trip-item">
+      <h5 className="trip-item-date">{newDate}</h5>
+      <div className="text-container">
+        <span className="paragraph">
+          <FaTrain /> {trainType}
+        </span>
+        <p className="paragraph">Passengers: {passengers}</p>
+        <p className="paragraph">CO2: {carbon} kg </p>
+        <p className="paragraph">{distance} km </p>
```
Comment (hugh-burgess) [inline src/components/TrainTripItem.js:31]: these too?

**Review by hugh-burgess — CHANGES_REQUESTED** [review]
Hey man, rocking it! Just a couple things for you CO2 look at

**Review by marwinburesch — COMMENTED** [review]
I see a lot of improvements
There could be more components and less repeating code, other than that I think you're on your way
You've added a lot this week, please make more PRs and commits
🐜

**Pull request merged**